### PR TITLE
feat: allow "client_secret" auth method with ADFS identity provider

### PR DESCRIPTION
**Pull request by t-chappl**
**Reason for Change**:

Allows a user to deploy kubernetes on Azure Stack with ADFS using a service principle client secret instead of a certificate. 

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_customcloud.sh [hunk 1]
```diff
@@ -58,17 +58,17 @@ configureK8sCustomCloud() {
         KUBERNETES_FILE_DIR=$(dirname "${AZURE_JSON_PATH}")
         K8S_CLIENT_CERT_PATH="${KUBERNETES_FILE_DIR}/k8s_auth_certificate.pfx"
         echo $SERVICE_PRINCIPAL_CLIENT_SECRET_CERT | base64 --decode > $K8S_CLIENT_CERT_PATH
-        # shellcheck disable=SC2002
-        cat "${AZURE_JSON_PATH}" | \
+        # shellcheck disable=SC2002,SC2005
+        echo $(cat "${AZURE_JSON_PATH}" | \
```
Comment (mboersma) [inline parts/k8s/cloud-init/artifacts/cse_customcloud.sh]: Is this an unrelated bug fix?

Comment (mboersma) [reply parts/k8s/cloud-init/artifacts/cse_customcloud.sh]: Because I changed this in #1230 (and tested the commands individually), but if so: sorry!

Comment (honcao) [reply parts/k8s/cloud-init/artifacts/cse_customcloud.sh]: @mboersma  It is ok , Because Azure Stack does not have the pipeline in the PR , we are working on that. hopefully we can get the it done in next month.

yes, it is a bug.

File: examples/azurestack/kubernetes-azurestack-adfs-secret.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "orchestratorRelease": "1.11",
+            "kubernetesConfig": {
+                "kubernetesImageBase": "msazurestackdocker/",
+                "useInstanceMetadata": false,
+                "networkPlugin": "flannel"
+            }
+        },
+        "customCloudProfile": {
+            "identitySystem": "adfs",
+            "authenticationMethod": "client_certificate",
```
Comment (t-chappl) [inline examples/azurestack/kubernetes-azurestack-adfs-secret.json]: This needs to be `client_secret`

File: parts/k8s/cloud-init/artifacts/cse_customcloud.sh [hunk 3]
```diff
@@ -58,17 +58,17 @@ configureK8sCustomCloud() {
...
-            jq 'del(.aadClientSecret)' > ${AZURE_JSON_PATH}
+            jq 'del(.aadClientSecret)') > ${AZURE_JSON_PATH}
     fi
 
     if [[ "${IDENTITY_SYSTEM,,}" == "adfs"  ]]; then
         # update the tenent id for ADFS environment.
-        # shellcheck disable=SC2002
-        cat "${AZURE_JSON_PATH}" | jq '.tenantId = "adfs"' > ${AZURE_JSON_PATH}
+        # shellcheck disable=SC2002,SC2005
+        echo $(cat "${AZURE_JSON_PATH}" | jq '.tenantId = "adfs"') > ${AZURE_JSON_PATH}
```
Comment (CecileRobertMichon) [inline parts/k8s/cloud-init/artifacts/cse_customcloud.sh]: this change might require you to push generated files. You can run `make generate` to generate those.
